### DISCUSSION TITLE
COBALT diagnostics for PAR, MLD, euphotic depth (Zeu)

### DISCUSSION THREAD

**Discussion by gmanville** [NOAA-GFDL/CEFI-regional-MOM6#80] [Q&A] [answered]
Hi all,

I'm new to COBALT and trying to implement a diagnostic seawater dimethylsulfide (DMS) parameterisation algorithm, for which I need a few input variables.

Can anyone please suggest the best diagnostics for MLD, PAR, and Zeu (1% penetration of PAR) in COBALT? Here's what I'm thinking:

PAR: output sfc_irrad (i,j) as a diagnostic? Or use cobalt%irr_inst(i,j,1) or cobalt%f_irr_aclm_sfc(i,j,1)?
MLD: cobalt%mld_aclm (i,j)?
Zeu: use cobalt%irr_inst(i,j,k), cobalt%f_irr_aclm_z (i,j,k), or cobalt%f_irr_aclm (i,j,k) to calculate depth at which values are 1% of surface?

I'd be grateful if anyone could point me in the right direction.

Thanks in advance!
George

**Comment by yichengt900** [comment]
Hi @gmanville,

I believe @charliestock and others can provide more details and additional insights, but here I just wanted to clarify the differences among `sfc_irrad(i,j)`, `cobalt%irr_inst(i,j,1)`, and `cobalt%f_irr_aclm_sfc(i,j,1)` so it might help you identify the most suitable input variables for your DMS parameterization algorithm:

`sfc_irrad(i,j)`: PAR at the air-sea interface. Keep in mind nn COBALT, We include all visible wavelengths, even those below 350-400 nm for PAR. Then the `par_adj` parameter, with a default value of 0.83, adjusts this to make PAR 47% of the total shortwave flux, aligning with Baker and Frouin (1987).

`cobalt%irr_inst(i,j,1)`: the instantaneous irradiance at the mid-point of the first grid cell.  It adjusts `sfc_irrad(i,j)` for water opacity and depth (half the thickness of the first grid cell, which I think is 1 meter in our case).

`cobalt%f_irr_aclm_sfc(i,j,1)`: is  the acclimation irradiance at the surface (calculated using src_irrad(i,j) and adjusted toward the current value with an inverse time scale). Essentially, it is an effective average of the surface instantaneous irradiance over daylight hours across an acclimation timescale, typically 24 hours.

And I think the calculation of `Zeu` will depend on whether you choose to use an instantaneous or acclimation irradiance. Keep in mind in COBALT  former is eventually used to calculate instantaneous phytoplankton growth, while the latter is used to calculate the chlorophyll to carbon ratio.

I hope this clarifies some of your questions and assists you in designing your DMS parameterization algorithm.

**Reply by gmanville** [reply]
Hi @yichengt900, 

Thanks for your help with this, your clarification is very helpful. I am adapting a semi-empirical algorithm that uses satellite PAR 1-8 day composites supplemented with climatological PAR, so I think all these model input options will offer more precision! I will speak to Charlie about whether he recommends instant or acclimation irradiance. For now, I'm using cobalt%irr_inst(i,j,1). 

Can you clarify that cobalt%mld_aclm (i,j) is the right MLD variable to be using? Is this the same as the MLD_003 I see in some output files?

**Comment by yichengt900** [comment, marked as answer]
@gmanville, I'm glad you found that helpful. Currently, `cobalt%mld_aclm(i,j)` is not exactly the same as MOM6's `MLD_003`, though the difference is minimal. This value is calculated within COBALT, using a default criterion based on de Boyer Montegut et al. (2004). The mixed layer depth is determined relative to a reference depth of 10m, with a potential density difference of 0.03 kg m-3. We have a [PR](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/pull/88) under review to incorporate the MLD directly from MOM6. For now, I think it's fine to continue using `cobalt%mld_aclm(i,j)` for your development.

**Reply by gmanville** [reply]
Ok great, thanks for your help @yichengt900.
